code/game/machinery/deployable.dm

```diff
@@ -290,16 +290,16 @@ for reference:
 		icon_state = "barrier[locked]"
 
 /obj/machinery/deployable/barrier/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)//So bullets will fly over and stuff.
-
 	if(istype(mover,/obj/item/projectile))
 		return (check_cover(mover,target))
 
 	if(air_group || (height==0))
-		return 1
-	if(istype(mover) && mover.checkpass(PASSTABLE))
-		return 1
-	else
-		return 0
+		return TRUE
+
+	if(ishuman(mover))
+		var/mob/living/carbon/human/H = mover
+		if(H.checkpass(PASSTABLE) && H.stats?.getPerk(PERK_PARKOUR))
+			return TRUE
```

Comment:
Do you need a return FALSE after this?

Reply:
Why? If no conditions for `return TRUE` is met, `null` would be returned, which is the same as `FALSE` for the purpose of `CanPass()` check.